internal/image_template.go

```diff
@@ -33,8 +34,25 @@ func (t *ImageTemplate) String() string {
 	return t.Raw
 }
 
+func prepTemplateString(input string) string {
+	output := ""
+	matches := TemplateRegex.FindAllStringSubmatchIndex(input, -1)
+	start := 0
+	for _, match := range matches {
+		output += input[start:match[2]] + "index ."
+		parts := strings.Split(input[match[2]+1:match[3]], ".")
```

Comment:
Would you mind in another PR to add some comments on what these lines do? The method seems quite obscure and difficult to grasp, at least to me.